File: annotations-runtime/src/main/java/com/github/dynamicextensionsalfresco/quartz/GenericQuartzJob.java
package com.github.dynamicextensionsalfresco.quartz;

import java.util.Map;
import org.quartz.Job;
import org.quartz.JobDetail;
import org.quartz.JobExecutionContext;
import org.quartz.JobExecutionException;

public final class GenericQuartzJob implements Job {

    public void execute(JobExecutionContext jobExecutionContext) throws JobExecutionException {
        JobDetail jobDetail = jobExecutionContext.getJobDetail();
        Map jobDataMap = jobDetail.getJobDataMap();
        Object obj = jobDataMap.get(QuartzJobRegistrar.BEAN_ID);
        Job lockedJob = (Job)obj;
        lockedJob.execute(jobExecutionContext);
    }

}

File: annotations-runtime/src/main/kotlin/com/github/dynamicextensionsalfresco/quartz/GenericQuartzJob.kt
package com.github.dynamicextensionsalfresco.quartz

import org.quartz.Job
import org.quartz.JobExecutionContext
import org.quartz.JobExecutionException

/**
 * Quartz requires a Job class, not an instance, so we use this generic delegate.

 * @author Laurent Van der Linden
 */
public class GenericQuartzJob : Job {

    @Throws(JobExecutionException::class)
    override fun execute(jobExecutionContext: JobExecutionContext) {
        val lockedJob = jobExecutionContext.getJobDetail().getJobDataMap().get(BEAN_ID) as Job
        lockedJob.execute(jobExecutionContext)
    }

    companion object {
        public val BEAN_ID: String = "bean"
    }
}
